client/src/components/RatingsAndReviews/AddReviewForm.jsx

```diff
@@ -0,0 +1,418 @@
+                    accept="image/png, image/jpeg"
+                    multiple
+                    onChange={this.handleFileUpload}
+                  />
+                </label>
+                <div>
+                  <button type="submit" value="upload">Upload</button>
+                </div>
+              </div> */}
+              <div className="review-user-nickname">
+                <label htmlFor="review-user-nickname">
+                  <div className="asterisk-wrapper">
+                    What is your nickname?
+                    <span className="required-review-mark">
+                      <i className="fas fa-asterisk" />
+                    </span>
+                  </div>
+                  <input
+                    type="text"
+                    id="review-user-nickname"
+                    name="name"
+                    required
+                    maxLength="60"
+                    size="20"
+                    placeholder="Example: jackson11!"
+                    onChange={this.handleInputChange}
+                  />
+                </label>
+                <div className="muted-reminders text-reminder">
+                  For privacy reasons, do not use your full name or email address
+                </div>
+              </div>
+              <div className="review-user-email">
+                <label htmlFor="review-user-email">
+                  <div className="asterisk-wrapper">
+                    What is your email?
+                    <span className="required-review-mark">
+                      <i className="fas fa-asterisk" />
+                    </span>
+                  </div>
+                  <input
+                    type="email"
+                    id="review-user-email"
+                    name="email"
+                    required
+                    maxLength="60"
+                    size="30"
+                    placeholder="Example: jackson11@email.com"
+                    onChange={this.handleInputChange}
+                  />
+                </label>
+                <div className="muted-reminders text-reminder">
+                  For authentication reasons, you will not be emailed
+                </div>
+              </div>
+            </div>
+            <div className="submit-button-holder">
+              <input
+                type="submit"
+                value="Submit Review"
+                className="interactive-button"
+              />
+              {ratingWarning}
+            </div>
+          </form>
+        </div>
+      </div>
+    );
+  }
+};
+
+AddReviewForm.propTypes = {
+  productInfo: PropTypes.instanceOf(Object).isRequired,
+  characteristics: PropTypes.instanceOf(Object).isRequired,
+  showAddReviewModal: PropTypes.bool.isRequired,
+  closeReviewFormHandler: PropTypes.instanceOf(Function).isRequired,
+  getCurrProdData: PropTypes.instanceOf(Function).isRequired,
+};
+
+export default AddReviewForm;
```

Comment:
This itself is a whole app hahaha. Bet you are tired of writing 'onChange' and 'onClick' XD